revpi_mio.c

```diff
@@ -39,7 +39,9 @@ static int revpi_mio_cycle_dio(SDevice *dev, SMioDigitalRequestData *req_data,
 
 	ret = pibridge_req_io(dev->i8uAddress, IOP_TYP1_CMD_DATA,
 			      &req, sizeof(req), &resp, sizeof(resp));
-	if (ret) {
+	if (ret != sizeof(resp)) {
+		if (ret > 0)
+			ret = -EIO;
 		pr_err_ratelimited("talk with mio for dio data error(addr:%d, "
 				   "ret:%d)\n", dev->i8uAddress, ret);
 		return ret;
```

Comment:
In the MIO case, `ret` is passed up the call chain but it's only interpreted by <a href="https://github.com/RevolutionPi/piControl/blob/ab3b62822abed8933ffeaa1a7e6eef55768cb062/RevPiDevice.c#L152">revpi_dev_update_state()</a>, which only checks whether it's non-zero. So changing to `-EIO` doesn't actually change the outcome. Of course, there could be other reasons to change the return value (taste, making things foolproof, ...).